### DISCUSSION TITLE
No support for key event input (eg; enter) to make editable row and/or new insert row from tables

### DISCUSSION THREAD

**Discussion by eisenhaus335** [Budibase/budibase#3778] [Ideas]
Hi, i just started out learning budibase. Looking on the documentation budibase doesnt has at all way to read key event
I figured to want make editable table that automatically save the row when triggered by defined event (onEnter). Is there any documentation that i missing out or this was designed like this to not listening client event (except onclick)?

If this the case. Adding a new kind of field group (so it will not clumped with other) will be helpful, that listen the application state so when client has done finished the field, it automatically insert new row and reset the state, client able to repeat the process again or close the screen. That will make input many field easier than smash the "save row" too many times like this https://github.com/Budibase/budibase/discussions/3221

Idk if this will bloated the library as this new kind of event listening might bad for performance or this new feature maybe far complex than i thought. But i am not sure, just for brainstorming idea. I am not that well versed with web performance. Hope this be helpful

**Comment by v3ss0n** [comment]
Key event handling are very important . Shouldn't it be easy to integrate inside Actions ?

**Reply by melohagan** [reply]
I do like the idea! Custom components are on the road map for this quarter, so hopefully that's something that could be built out.
